internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+)
+
+type manifest struct {
+	Owner struct {
+		Username string `json:"username"`
+		Name     string `json:"name"`
+	} `json:"owner"`
+	Name    string `json:"name"`
+	Title   string `json:"title"`
+	Version string `json:"version"`
+	Archive struct {
+		Url  string `json:"url"`
+		Md5  string `json:"md5"`
+		Sha1 string `json:"sha1"`
+	} `json:"archive"`
+	License []struct {
+		Name string `json:"name"`
+		Url  string `json:"url"`
+	} `json:"license"`
+}
+
+const (
+	archiveInstallation = "ARCHIVE"
+	packageInstallation = "PACKAGE"
+
+	invalidDirectoryErrorMsg       = `plugin directory "%s" does not exist`
+	unexpectedInstallationErrorMsg = "unexpected installation type: %s"
+)
+
+func (c *pluginCommand) newInstallCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install <plugin>",
+		Short: "Install a Connect plugin.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.install,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Install the latest release of the Datagen connector onto your local Confluent Platform environment.",
+				Code: "confluent connect plugin install confluentinc/kafka-connect-datagen:latest",
+			},
+		),
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireOnPremLogin},
+	}
+
+	cmd.Flags().String("plugin-directory", "", "The plugin installation directory.")
+	cmd.Flags().StringSlice("worker-configs", []string{}, "A comma-separated list of paths to one or more Kafka Connect worker configuration files. Each worker file will be updated to load plugins from the plugin directory in addition to any preexisting directories.")
+	cmd.Flags().Bool("dry-run", false, "Simulate an operation without making any changes.")
+	cmd.Flags().Bool("force", false, "Proceed without user input.")
+
+	return cmd
+}
+
+func (c *pluginCommand) install(cmd *cobra.Command, args []string) error {
+	dryRun, err := cmd.Flags().GetBool("dry-run")
+	if err != nil {
+		return err
+	}
+	if dryRun {
+		output.Println("[DRY RUN] Performing a dry run of this command.")
+	}
+	// Verify that the argument corresponds to a valid plugin
+	var pluginManifest *manifest
+	if utils.DoesPathExist(args[0]) {
+		// if installing plugin from local archive
+		localManifest, err := getLocalManifest(args[0])
+		if err != nil {
+			return err
+		}
+		pluginManifest = localManifest
+	} else {
+		// if installing plugin from Confluent Hub
+		owner, name, version, err := parsePluginId(args[0])
+		if err != nil {
+			return err
+		}
+
+		remoteManifest, err := getRemoteManifest(owner, name, version)
+		if err != nil {
+			return err
+		}
+		pluginManifest = remoteManifest
+	}
+
+	// Select plugin-directory
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+
+	pluginDir, err := getPluginDirFromFlag(cmd)
+	if err != nil {
+		return err
+	}
+
+	var ins *installation
```

Comment:
Let's find a better name for this since it has a really long scope, and `ins` isn't super descriptive.